pkg/sfu/streamallocator.go

```diff
@@ -695,36 +705,52 @@ func (s *StreamAllocator) handleNewEstimateInProbe() {
 		//
 		// More of a safety net.
 		// In rare cases, the estimate gets stuck. Prevent from probe running amok
-		// LK-TODO: Need more testing this here and ensure that probe does not cause a lot of damage
+		// LK-TODO: Need more testing here to ensure that probe does not cause a lot of damage
 		//
 		s.params.Logger.Debugw("probe: aborting, no trend")
 		s.abortProbe()
-	case trend == EstimateTrendDownward:
-		// stop immediately if estimate falls below the previously committed estimate, the probe is congesting channel more
-		s.params.Logger.Debugw("probe: aborting, estimate is trending downward")
+	case trend == ChannelTrendCongesting:
+		// stop immediately if the probe is congesting channel more
+		s.params.Logger.Debugw("probe: aborting, channel is congesting")
 		s.abortProbe()
-	case s.probeEstimator.GetHighest() > s.probeGoalBps:
+	case s.probeChannelObserver.GetHighestEstimate() > s.probeGoalBps:
 		// reached goal, stop probing
 		s.params.Logger.Debugw("probe: stopping, goal reached")
 		s.stopProbe()
 	}
 }
 
 func (s *StreamAllocator) handleNewEstimateInNonProbe() {
-	trend := s.estimator.AddEstimate(s.lastReceivedEstimate)
-	if trend != EstimateTrendDownward {
+	s.channelObserver.AddEstimate(s.lastReceivedEstimate)
+
+	packetDelta, repeatedNackDelta := s.getNackDelta()
+	s.channelObserver.AddNack(packetDelta, repeatedNackDelta)
+
+	trend := s.channelObserver.GetTrend()
+	if trend != ChannelTrendCongesting {
+		return
+	}
+
+	nackRatio := s.channelObserver.GetNackRatio()
+	lossAdjustedEstimate := s.lastReceivedEstimate
+	if nackRatio > NackRatioThresholdNonProbe {
+		lossAdjustedEstimate = int64(float64(lossAdjustedEstimate) * (1.0 - NackRatioAttenuator*nackRatio))
+	}
+	if s.committedChannelCapacity == lossAdjustedEstimate {
```

Comment:
This is where we apply the NACK ratio attenuator to ratchet down the committed estimate.